test/lib/mayaUsd/render/mayaToHydra/cpp/testFlowViewportAPIFilterPrims.cpp

```diff
@@ -0,0 +1,145 @@
+    //This is the case for the object named "bigSphere", it has more than 10 000 vertices.
+    Fvp::FilteringSceneIndexClientExample hydraViewportFilteringSceneIndexExample ("TestFilteringSceneIndex", 
+                                                        Fvp::FilteringSceneIndexClient::Bucket::kSceneFiltering,
+                                                        FvpViewportAPITokens->allRenderers,
+                                                        &parentShapeMOject //Set the cube as the parent of this filtering scene index
+                                                        );
+    //Filtering scene index interface
+    Fvp::FilteringSceneIndexInterface& filteringSceneIndexInterface = Fvp::FilteringSceneIndexInterface::get();
+    
+    //Register it
+    const bool bRes = filteringSceneIndexInterface.registerFilteringSceneIndexClient(hydraViewportFilteringSceneIndexExample);
+    ASSERT_TRUE(bRes);
+
+    //Check that there are primitives in the viewport terminal scene index
+    const SceneIndicesVector& sceneIndices = GetTerminalSceneIndices();
+    ASSERT_GT(sceneIndices.size(), static_cast<size_t>(0));
+    SceneIndexInspector inspector(sceneIndices.front());
+
+    // Retrieve the small sphere prim in the list of primitives and return its visibility
+    FindPrimPredicate findSmallSpherePrimPredicate
+        = [](const HdSceneIndexBasePtr& sceneIndex, const SdfPath& primPath) -> bool {
+        const std::string primPathString    = primPath.GetAsString();
+        HdSceneIndexPrim prim               = sceneIndex->GetPrim(primPath);
+        if (primPathString.find(smallSphereShapeName.asChar()) != std::string::npos) {
+            //Check if it is visible or not
+            auto visibilityHandle = HdVisibilitySchema::GetFromParent(prim.dataSource).GetVisibility();
+            if (visibilityHandle){
+                return visibilityHandle->GetTypedValue(0.0f); //return true if it is visible, false otherwise
+            }
+        }
+        return false;
+    };
+
+    // Retrieve the big sphere prim in the list of primitives and return its visibility
+    FindPrimPredicate findBigSpherePrimPredicate
+        = [](const HdSceneIndexBasePtr& sceneIndex, const SdfPath& primPath) -> bool {
+        const std::string primPathString    = primPath.GetAsString();
+        HdSceneIndexPrim prim               = sceneIndex->GetPrim(primPath);
+        if (primPathString.find(bigSphereShapeName.asChar()) != std::string::npos) {
+            //Check if it is visible or not
+            auto visibilityHandle = HdVisibilitySchema::GetFromParent(prim.dataSource).GetVisibility();
+            if (visibilityHandle){
+                return visibilityHandle->GetTypedValue(0.0f); //return true if it is visible, false otherwise
+            }
+        }
+        return false;
+    };
```

Comment:
Duplicate, might have done better with a class and bool operator():
```suggestion
    struct ShapeVisibilityPredicate {
        const MString _name;
        ShapeVisibilityPredicate(const MString& name) : _name(name) {}
        bool operator()(const HdSceneIndexBasePtr& sceneIndex, const SdfPath& primPath) {
            const std::string primPathString    = primPath.GetAsString();
            HdSceneIndexPrim prim               = sceneIndex->GetPrim(primPath);
            if (primPathString.find(_name.asChar()) != std::string::npos) {
                //Check if it is visible or not
                auto visibilityHandle = HdVisibilitySchema::GetFromParent(prim.dataSource).GetVisibility();
                if (visibilityHandle){
                    return visibilityHandle->GetTypedValue(0.0f); //return true if it is visible, false otherwise
                }
            }
            return false;
        }
    };
```

Reply:
Thanks, I've added a class in testutils.h and used it in different tests.